File: pynagent/tests/fixtures/java/vulns.java
// Test fixtures for Java security rules
// DO NOT use as templates!

import java.io.*;
import javax.crypto.*;

public class VulnerableCode {

    // Hardcoded Secrets
    private static final String API_KEY = "sk_live_abcdef1234567890abcdef1234567890";
    private static final String PASSWORD = "SuperSecret123!";

    // SQL Injection
    public void unsafeQuery(String userId) throws Exception {
        String query = "SELECT * FROM users WHERE id=" + userId;
        Statement stmt = connection.createStatement();
        ResultSet rs = stmt.executeQuery(query);
    }

    // Command Injection
    public void runCommand(String cmd) throws Exception {
        Runtime.getRuntime().exec(cmd);
        ProcessBuilder pb = new ProcessBuilder(userInput);
        pb.start();
    }

    // Path Traversal
    public String readFile(String filename) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader(userFilename));
        return reader.readLine();
    }

    // Weak Crypto
    public void weakCrypto() throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        Cipher des = Cipher.getInstance("DES");
    }

    // Debug statements
    public void processOrder(String order) {
        System.out.println("Processing order: " + order);
    }
}

File: pynagent/tests/fixtures/csharp/vulns.cs
// Test fixtures for C# security rules
// DO NOT use as templates!

using System;
using System.Diagnostics;
using System.IO;
using System.Web;
using System.Security.Cryptography;

public class VulnerableCode
{
    // Hardcoded Secrets
    private const string ApiKey = "sk_live_abcdef1234567890";
    private const string Password = "SuperSecret123!";

    // SQL Injection
    public void UnsafeQuery(string userId)
    {
        var query = "SELECT * FROM users WHERE id=" + userId;
        var cmd = new SqlCommand(query, conn);
    }

    // Command Injection
    public void RunCommand(string cmd)
    {
        Process.Start(userInput);
        var psi = new ProcessStartInfo("cmd.exe", "/c " + userCommand);
        Process.Start(psi);
    }

    // Path Traversal
    public string ReadFile(string filename)
    {
        return File.ReadAllText(userFilename);
    }

    // Weak Crypto
    public void WeakCrypto()
    {
        using (var md5 = MD5.Create())
        using (var sha1 = SHA1.Create())
        using (var des = DES.Create())
        {
        }
    }

    // Debug statements
    public void ProcessOrder(string order)
    {
        Console.WriteLine("Processing order: " + order);
    }

    // XSS
    public void RenderContent(string content)
    {
        Response.Write(userContent);
    }
}
